### PR TITLE
Notifications: Added else if clause for users with both email & phone login method

### PR DESCRIPTION
**Pull request by abhinavjonnada82**
This PR addresses following issue:
https://github.com/episphere/connect/issues/689

When notifications (esp. emails) are sent out for users with both email & phone login method. The `<loginDetails>` in email is shown as undefined. To prevent this added else if clause to capture users with both login methods.

- Modularized functions that redact email & phone. 
- Returns following statements:
    Email Login: `Your login information for the MyConnect app is ` + `email` + `.`;
    Phone Login: `Your login information for the MyConnect app is` + `phone` + `.`;
    Both Email & Phone Login: `Your login information for the MyConnect app is ` + `email` + `or` + `phone` + `.`;
    
PoC:

Used a workaround to see how the text renders:
  
Phone:
Screenshot 2023-08-17 at 4 02 44 PM

Email:
Screenshot 2023-08-17 at 4 38 23 PM

Both Phone & Email:
Screenshot 2023-08-17 at 4 40 18 PM

### CODE REVIEW

File: utils/shared.js [hunk 1]
```diff
@@ -757,6 +757,18 @@ const isDateTimeFormat = (value) => {
...
 }
 
+const redactEmailLoginInfo = (loginDetails) => {
+    let amp = loginDetails.indexOf('@');    
+    for(let i = 0; i < amp; i++) {
+        if(i != 0 && i != 1 && i != amp - 1) {
+            loginDetails = loginDetails.substring(0, i) + "*" + loginDetails.substring(i + 1);
+        } 
+    }
+    return loginDetails
```
Comment (Gbarra9) [inline utils/shared.js:767]: Nice refactor for the email and phone redacting logic. First time seeing a the redacting of characters for email address.Interesting way of redacting the characters before the "@" in the email address.

**Review by Gbarra9 — APPROVED** [review]
Looks good

File: utils/shared.js (hunk 1 above)
Comment (JoeArmani) [inline utils/shared.js:767]: Looks good, nice way to redact an email address!
Here's an example that also handles a very short email address and avoids rebuilding the string every iteration:
```javascript
const redactEmailLoginInfo = (loginDetails) => {
    const amp = loginDetails.indexOf('@');

    if (amp < 3) return loginDetails; // corner case (first two chars + last char before '@')
    
    let redacted = loginDetails.substring(0, 2); // Keep the first two characters
    
    for (let i = 2; i < amp - 1; i++) {
        redacted += '*';
    }

    redacted += loginDetails.charAt(amp - 1);
    redacted += loginDetails.substring(amp);
    
    return redacted;
};
```

Comment (we-ai) [reply utils/shared.js:767]: I'm doing the same part for speed. And avoided "for" loop:
```js
  const [prefix, domain] = fetchedData[421823980].split("@");
  const changedPrefix = prefix.length > 3
      ? prefix.slice(0, 2) + "*".repeat(prefix.length - 3) + prefix.slice(-1)
      : prefix.slice(0, -1) + "*";
  loginDetails = changedPrefix + "@" + domain;
```
We do have participants whose emails only have 2 letters beside @ and domain.

File: utils/shared.js [hunk 2]
```diff
@@ -757,6 +757,18 @@ const isDateTimeFormat = (value) => {
...
+}
+
+const redactPhoneLoginInfo = (loginDetails) => { return loginDetails = "***-***-" + loginDetails.substring(loginDetails.length - 4); }
```
Comment (we-ai) [inline utils/shared.js:770]: I used `loginDetails = "***-***-" + fetchedData[348474836].slice(-4);`

**Review by we-ai — APPROVED** [review]
Looks good.

File: utils/shared.js (hunk 1 above)
Comment (abhinavjonnada82) [reply utils/shared.js:767]: Thanks. The corner case is taken care in the original code. I leave it up to @anthonypetersen if he wants to change the function or not.

File: utils/shared.js (hunk 2 above)
Comment (abhinavjonnada82) [reply utils/shared.js:770]: This is something simpler tho.

File: utils/shared.js (hunk 1 above)
Comment (abhinavjonnada82) [reply utils/shared.js]: https://github.com/episphere/connectFaas/pull/399/files#diff-23fab349744c189c5b655022c3edc921526e74945e34cdd65db82880911119d8R760

**Review by we-ai — APPROVED** [review]
Looks good.

**Review by JoeArmani — APPROVED** [review]
👍

**Pull request merged**